src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -365,6 +359,42 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
   // Internal state, for memoization purposes only
   const [walletType, setWalletType] = useState<KeyManagerWithProvider | null>(null)
 
+  const getAdapter = useCallback(
+    async (keyManager: KeyManager, index: number = 0) => {
+      let currentStateAdapters = state.adapters ?? new Map<KeyManager, any[]>()
```

Comment:
yeah we should have a type that can map a `KeyManager` to an adapter such that casting isnt required everywhere

Reply:
Discussed this one out, will leave it for a part 4 PR in the stack since this will require an hdwallet bumpy